servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultMultiAddressUrlHttpClientBuilder.java

```diff
@@ -135,6 +135,9 @@ public StreamingHttpClient buildStreaming() {
             urlClient = redirectConfig == null ? urlClient :
                     new RedirectingHttpRequesterFilter(redirectConfig).create(urlClient);
 
+            // Detect leaks that can be caused by unexpected exceptions
+            urlClient = HttpMessageDiscardWatchdogClientFilter.CLIENT_CLEANER.create(urlClient);
```

Comment:
I think this results in 2 cleaners. Maybe it's not a big deal, but just wanted to be sure I understood what the structure will be.

Reply:
Correct, because from a single address client we don't know if it's a standalone instance or part of the multi-address client group we can not remove it from there. So, for multi-address flow we will have 2 cleaners for `onError` case:
1. Last one before returning from single-address level.
2. Last one before returning from multi-address level to the users.

This should not be a big deal because the watchdog overhead mostly comes from the connection level filter that instantiates `AtomicReference` for each request and transforms response message body for each response. The overhead of `CleanerStreamingHttpClientFilterFactory` is negligible bcz `onError` handling is super cheap and happens only in case of exceptions.

Reply:
Sent a new commit to also replace `onErrorResume` with `whenOnError`. It will help to reduce overhead (avoids subscribe sequence for returned failed single).

Reply:
I was suggesting we could make the Single client aware if it gets instantiated from a multi client, but your explanation makes me think it's probably not a big deal. Let's be sure and check the perf stats after merge.